hwy/tests/mask_mem_test.cc

```diff
@@ -109,6 +111,62 @@ HWY_NOINLINE void TestAllMaskedScatter() {
   ForUIF3264(ForPartialVectors<TestMaskedScatter>());
 }
 
+struct TestScatterIndexN {
+  template <class T, class D>
+  HWY_NOINLINE void operator()(T /*unused*/, D d) {
+    RandomState rng;
+
+    using TI = MakeSigned<T>;  // For mask > 0 comparison
+    const Rebind<TI, D> di;
+    using VI = Vec<decltype(di)>;
+    const size_t N = Lanes(d);
+    auto lanes = AllocateAligned<T>(N);
+    auto expected = AllocateAligned<T>(N);
+    HWY_ASSERT(lanes && expected);
+
+    const Vec<D> v = Iota(d, static_cast<T>(hwy::Unpredictable1() - 1));
+    Store(v, d, lanes.get());
+
+    const VI indices =
+        Reverse(di, Iota(di, static_cast<TI>(hwy::Unpredictable1() - 1)));
+
+    for (size_t rep = 0; rep < AdjustedReps(200); ++rep) {
+      // Choose 1 to N lanes to store
+      const size_t max_lanes_to_store = (Random32(&rng) % N) + 1;
+
+      ZeroBytes(expected.get(), N * sizeof(T));
+      for (size_t i = 0; i < max_lanes_to_store; ++i) {
+        expected[N - 1 - i] = static_cast<T>(i);
+      }
+
+      ZeroBytes(lanes.get(), N * sizeof(T));
+      ScatterIndexN(v, d, lanes.get(), indices, max_lanes_to_store);
+      HWY_ASSERT_VEC_EQ(d, expected.get(), Load(d, lanes.get()));
+    }
+
+    // Zero store is just zeroes
+    ZeroBytes(expected.get(), N * sizeof(T));
+    ZeroBytes(lanes.get(), N * sizeof(T));
+    ScatterIndexN(v, d, lanes.get(), indices, 0);
+    HWY_ASSERT_VEC_EQ(d, expected.get(), Load(d, lanes.get()));
+
+    // Load is clamped at min(N, max_lanes_to_load)
+    auto larger_memory = AllocateAligned<T>(N * 2);
+    auto larger_expected = AllocateAligned<T>(N * 2);
```

Comment:
Would you mind adding `HWY_ASSERT(larger_memory && larger_expected)` here for safety? IIRC this was required to satisfy the static analysis on one of the platforms.

Reply:
Done 😸